bpf/bpf_sock.c

```diff
@@ -1216,6 +1261,30 @@ int cil_sock6_getpeername(struct bpf_sock_addr *ctx)
 }
 #endif /* ENABLE_SOCKET_LB_PEER */
 
+__section("cgroup/sock_release")
+int cil_sock_release(struct bpf_sock *ctx __maybe_unused) {
+
+    #ifdef ENABLE_IPV4
+    if (ctx->family == AF_INET) {
+        if (sock4_delete_revnat(ctx) < 0) {
+            update_metrics(0, METRIC_EGRESS, REASON_LB_REVNAT_DELETE);
```

Comment:
I think we'll want to update the metric when the deletion succeeds rather than when it fails.